enhancements/network/ovn-k-localnet-api.md

```diff
@@ -0,0 +1,514 @@
+| Role                | Select the network role in the pod.                                                                                                                                                                                                                                                                   | No       |
+| PhysicalNetworkName | The OVN bridge mapping network name is configured on the node.                                                                                                                                                                                                                                        | No       |
+| MTU                 | The maximum transmission unit (MTU).                                                                                                                                                                                                                                                                  | Yes      |
+| VLAN                | The network VLAN ID.                                                                                                                                                                                                                                                                                  | Yes      |
+| Subnets             | List of CIDRs used for the pod network across the cluster.Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed. The format should match standard CIDR notation (for example, "10.128.0.0/16"). This field must be omitted if `ipam.mode` is `Disabled`. | Yes      |
+| ExcludeSubnets      | List of CIDRs removed from the specified CIDRs in `subnets`.The format should match standard CIDR notation (for example, "10.128.0.0/16"). This field must be omitted if `subnetes` is unset or `ipam.mode` is `Disabled`.                                                                            | Yes      |
+| IPAM                | Contains IPAM-related configuration for the network. When ipam.lifecycle=Persistent enable workloads have persistent IP addresses. For example: Virtual Machines will have the same IP addresses along their lifecycle (stop, start migration, reboots).                                              | Yes      |
+
+#### Suggested API validations 
+- `Role`:
+  - Required.
+  - Can be Secondary only when topology=Localnet.
+    - Having Role explicitly makes the API predictable and consistent with other topologies. In addition it enables extending localnet for Primary networks.
+- `PhysicalNetworkName`: Required.
+- `MTU`:
+  - Minimum 576 (minimal for IPv4). Maximum: 65536.
+  - When Subnets consist of IPv6 CIDR, minimum MTU should be 1280.
+- `VLAN`:
+  - VLAN IDs `0` (untagged),`1` (default), highest VLAN ID is `4095`. 
+    Minimum: 2, Maximum: 4095.
```

Comment:
I think the highest is 4094 actually. 4095 is reserved. Check for instance https://networkengineering.stackexchange.com/questions/24404/vlan-0-1-and-4095-are-reserved-what-are-they-reserved-for

Reply:
Thanks, fixed. Also added ref.